core/trino-main/src/main/java/io/trino/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -171,20 +170,35 @@ public static PlanNodeStatsEstimate capStats(PlanNodeStatsEstimate stats, PlanNo
         return result.build().getSymbolStatistics();
     }
 
+    @FunctionalInterface
+    public interface PlanNodeStatsEstimateInfo
+    {
+        PlanNodeStatsEstimate getEstimate();
+    }
+
+    @FunctionalInterface
+    public interface SymbolStatsCombiner
+    {
+        Map<Symbol, SymbolStatsEstimate> intersect(Map<Symbol, SymbolStatsEstimate> combined, PlanNodeStatsEstimateInfo termEstimateInfo);
+    }
+
     public static PlanNodeStatsEstimate estimateCorrelatedConjunction(
             PlanNodeStatsEstimate input,
-            List<PlanNodeStatsEstimate> estimates,
+            List<PlanNodeStatsEstimateInfo> estimates,
             double independenceFactor,
-            BinaryOperator<Map<Symbol, SymbolStatsEstimate>> symbolStatsCombiner)
+            SymbolStatsCombiner symbolStatsCombiner)
```

Comment:
You don't need combiner if you pass combine stat estimates before calling `estimateCorrelatedConjunction`. It's also better for perf since you can combine them all together